src/clusterinfo.c

```diff
@@ -248,127 +244,112 @@ void loadbandmap(void) {
     pthread_mutex_unlock(&spot_ptr_mutex);
 
 
+    /* show last 8 spots via xplanet */
     linepos = (i < 8 ? 0 : i - 8);
 
-    /* prune markerfile by opening it for write */
-    if (xplanet > 0 && nofile == 0) {
-	if ((fp = fopen(markerfile, "w")) == NULL) {
-	    nofile = 1;			/* remember: no write possible */
-	    TLF_LOG_INFO("Opening marker file not possible.");
-	} else
-	    fclose(fp);
-    }
-
     for (j = linepos; j < linepos + 8; j++) {
 
 	if (bandmap[j] != NULL) {
-	    g_strlcpy(spotline, bandmap[j] + 17, 23);	// freq and call
-	    g_strlcpy(spottime, bandmap[j] + 70, 6);	// time
-	    strcat(spotline, spottime);
+	    char marker_out[60];
+	    char color[sizeof("Magenta")];
+	    int lon;
+	    int lat;
+	    int ctynr;
+	    bool iswarc;
 
 	    strncpy(callcopy, bandmap[j] + 26, 16);	// call
-	    for (m = 0; m < 16; m++) {
+	    for (int m = 0; m < 16; m++) {
 		if (callcopy[m] == ' ') {
 		    callcopy[m] = '\0';
 		    break;
 		}	/* use strcspn? */
 	    }
 
-	    x = getctynr(callcopy);		// CTY of station
-
-
-	    if (x != 0 && xplanet > 0 && nofile == 0) {
-
-		if ((fp = fopen(markerfile, "a")) == NULL) {
-		    TLF_LOG_INFO("Opening markerfile not possible.");
-		} else {
-
-		    /* show no callsign if MARKERDOTS */
-		    if (xplanet == 2)
-			callcopy[0] = '\0';
-
-		    dx = dxcc_by_index(x);
-		    lon = (int)(dx -> lon) * -1;
-		    lat = (int)(dx -> lat);
-
-		    *color = '\0';
-
-		    if (spot_age[j] > 15)
-			strcat(color, "Green");
-		    else {
-			iswarc = 0;
-			if (spot_freq[j] >= 10100.0 && spot_freq[j] <= 10150.0)
-			    iswarc = 1;
-			if (spot_freq[j] >= 18068.0 && spot_freq[j] <= 18168.0)
-			    iswarc = 1;
-			if (spot_freq[j] >= 24890.0 && spot_freq[j] <= 24990.0)
-			    iswarc = 1;
-
-			if (iswarc == 0) {
-			    if (spot_freq[j] < 3500.0)
-				strcat(color, "Red");
-			    if (spot_freq[j] >= 3500.0
-				    && spot_freq[j] <= 4000.0)
-				strcat(color, "Magenta");
-			    if (spot_freq[j] >= 7000.0
-				    && spot_freq[j] <= 7300.0)
-				strcat(color, "Yellow");
-			    if (spot_freq[j] >= 14000.0
-				    && spot_freq[j] <= 14350.0)
-				strcat(color, "Blue");
-			    if (spot_freq[j] >= 21000.0
-				    && spot_freq[j] <= 21450.0)
-				strcat(color, "White");
-			    if (spot_freq[j] >= 28000.0
-				    && spot_freq[j] <= 29700.0)
-				strcat(color, "Green");
-
-			} else {
-			    strcat(color, "Cyan");
-			}
-		    }
+	    ctynr = getctynr(callcopy);		// CTY of station
+
+	    if (ctynr != 0 ) {
+		/* show no callsign if MARKERDOTS */
+		if (xplanet == 2)
```

Comment:
when refactoring logcfg parsing I thought it'd make sense to have constants for Xplanet modes, just didn't want to mess with it